shark_turbine/kernel/wave/wave.py

```diff
@@ -197,10 +202,13 @@ def _trace_and_get_kernel_signature(
         exe = dispatch_codegen.StreamExecutable(mb, name=entrypoint_name)
         dispatch_entrypoint = exe.define_entrypoint(entrypoint_name, kernel_sig, grid)
 
-        emitter = WaveEmitter(dispatch_entrypoint, graph)
+        emitter = WaveEmitter(dispatch_entrypoint, graph, self.constraints)
         emitter.emit(graph.get_root_graph())
         emitter.finish()
 
+        if kwargs.get("canonicalize", False):
```

Comment:
When do we not want to canonicalize?

Reply:
You don't want to canonicalize when you will lose all your IR on canonicalization (because they have no uses). This happens on some of the other tests (you can try it out by setting canonicalize=True on some of the other tests).